src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,716 @@
+    {"dsFPD_LED_DEVICE_WPS_CONNECTING", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTING},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTED", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTED},
+    {"dsFPD_LED_DEVICE_WPS_ERROR", (int32_t)dsFPD_LED_DEVICE_WPS_ERROR},
+    {"dsFPD_LED_DEVICE_FACTORY_RESET", (int32_t)dsFPD_LED_DEVICE_FACTORY_RESET},
+    {"dsFPD_LED_DEVICE_USB_UPGRADE", (int32_t)dsFPD_LED_DEVICE_USB_UPGRADE},
+    {"dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR", (int32_t)dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR},
+    {"dsFPD_LED_DEVICE_MAX", (int32_t)dsFPD_LED_DEVICE_MAX},
+    {NULL, -1}
+};
+
+void readAndDiscardRestOfLine(FILE* in)
+{
+    int c;
+    while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the Device settings - Front Panel Module
+*
+* This test provides a scope to Intialize the Front Panel module.
+
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_1_dsFPD_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsFPInit()
+   UT_LOG_INFO("Calling dsFPInit()");
+   status = dsFPInit();
+   UT_LOG_INFO("Result dsFPInit(OUT:dsError_t:[%s])",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+   assert(status == dsERR_NONE);
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED state.
+*
+* This test case provides a scope to  set the State of FP indicator to ON and OFF.
+*
+* **Pre-Conditions:** @n
+* Front Panle Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 2 and provide indicator index and State to be set.
+*
+*/
+void test_2_dsFPD_hal_SetFPState(void)
+{
+    gTestID = 2;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    int32_t eIndicator;
+    int32_t indicatorType = 0;
+    uint32_t uState = 0;
+    int32_t dsFPDNumberOfIndicators = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+    char buffer[DS_FPD_KEY_SIZE];
+
+    UT_LOG_INFO(" \t  Supported Indicators are:");
+    UT_LOG_INFO("------------------------------------------");
+    for (int indicator = 1; indicator <= dsFPDNumberOfIndicators; indicator++) {
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", indicator);
+        indicatorType = UT_KVP_PROFILE_GET_UINT32(buffer);
+        UT_LOG_INFO("\t%d.  %-20s\n", indicatorType, UT_Control_GetMapString(dsFrontPanelIndicatorTable, indicatorType));
+    }
+    UT_LOG_INFO("----------------------------------------------------------");
+    UT_LOG_INFO("Select Indicator : ");
+    scanf("%d", &eIndicator);
+    readAndDiscardRestOfLine(stdin);
```

Comment:
Please check whether this can be defined in separate function. Since it is used repeatedly.

Reply:
done